src/storage/index/hash_index.cpp

```diff
@@ -99,10 +105,11 @@ HashIndex<T>::HashIndex(const DBFileIDAndName& dbFileIDAndName,
     const std::shared_ptr<BMFileHandle>& fileHandle, OverflowFileHandle* overflowFileHandle,
     uint64_t indexPos, BufferManager& bufferManager, WAL* wal)
     : dbFileIDAndName{dbFileIDAndName}, bm{bufferManager}, wal{wal}, fileHandle(fileHandle),
-      overflowFileHandle(overflowFileHandle) {
+      overflowFileHandle(overflowFileHandle), bulkInsertLocalStorage{overflowFileHandle} {
+    // TODO: Handle data not existing
     headerArray = std::make_unique<BaseDiskArray<HashIndexHeader>>(*fileHandle,
```

Comment:
Yes. I was thinking that we could probably store the headers in the PrimaryKeyIndex instead of in the HashIndex, so that we could write them as a single page (each header will need to be <=16 bytes). Most of the fields in the header can be calculated when they are read from disk.

I think all that's needed is:
- 1 byte for currrentLevel
- 8 bytes for numEntries

The masks are already just being calculated from `currentLevel`. nextSplitSlotId can be calculated from the number of primary slots, but that admittedly makes things more complicated as it's stored elsewhere. Alternatively I suppose they could be stored in two pages, which might be a little simpler.